src/gui/editors/AutomationEditor.cpp

```diff
@@ -1822,6 +1988,63 @@ AutomationEditor::timeMap::iterator AutomationEditor::getNodeAt(int x, int y, bo
 
 
 
+/**
+ * @brief Given a mouse X coordinate, returns a timeMap::iterator that points to
+ *        the closest node.
+ * @param Int X coordinate
+ * @return timeMap::iterator with the closest node or timeMap.end() if there are no nodes.
+ */
+AutomationEditor::timeMap::iterator AutomationEditor::getClosestNode(int x)
+{
+	// Remove the VALUES_WIDTH from the x position, so we have the actual viewport x
+	x -= VALUES_WIDTH;
+	// Convert the x position to the position in ticks
+	int posTicks = (x * TimePos::ticksPerBar() / m_ppb) + m_currentPosition;
+
+	// Get our pattern timeMap and create a iterator so we can check the nodes
+	timeMap & tm = m_clip->getTimeMap();
+
+	if (tm.isEmpty()) { return tm.end(); }
+
+	// Get the node with an equal or higher position
+	timeMap::iterator it = tm.lowerBound(posTicks);
```

Comment:
```suggestion
	auto it = tm.lowerBound(posTicks);
```
This is me nitpicking again